### ISSUE TITLE
RS1008 not reported on `ISymbol` fields of diagnostic analyzer

### ISSUE THREAD

**Issue by Rekkonnect** [dotnet/roslyn-analyzers#7196]
### Analyzer

**Diagnostic ID**: [RS1008](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs1008-avoid-storing-per-compilation-data-into-the-fields-of-a-diagnostic-analyzer): `Avoid storing per-compilation data into the fields of a diagnostic analyzer`

### Analyzer source

**Microsoft.CodeAnalysis.CSharp.Analyzers** 3.3.4

### Describe the bug

The above diagnostic is not reported on field instances of type `ISymbol`, but it is correctly reported on other relevant types like `IMethodSymbol`, `INamedTypeSymbol`, `Compilation`, etc.

### Example Code

Consider the following analyzer:
```csharp
[DiagnosticAnalyzer(LanguageNames.CSharp)]
public class TestAnalyzer : DiagnosticAnalyzer
{
    private ISymbol _symbol;

    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
        => ImmutableArray<DiagnosticDescriptor>.Empty;

    public override void Initialize(AnalysisContext context)
    {
        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.ReportDiagnostics);
        context.EnableConcurrentExecution();
    }
}
```

### Expected behavior

A warning should be reported on the type of the field `_symbol`, like it would if the field's type was `IMethodSymbol`, `INamedTypeSymbol`, etc.

### Actual behavior

The warning is not present.

**Comment by mpidash** [comment]
Fields of type `IOperation` are also not reported.
It seems that the analyzer only checks if the field type implements `IOperation` or `ISymbol`, but not if it is an exact match.

I will prepare a PR for this issue.
